### PR TITLE
replace sqlparser with lemon parser

### LINKED ISSUE #15: END statement not supported by parser
We use sqlparser-rs to parse and analyze queries. Despite using the SQLite dialect, compatibility with the language seems a bit limited. Support for the `END` statement seems to be one such instance of bad support.

Maybe we should consider alternatives. I'm considering https://github.com/gwenn/lemon-rs.

In the meantime, we should be returning a parsing error, rather than ignoring it.

### PR DESCRIPTION
**Pull request by MarinPostma**
The PR ~makes lemonade~ replaces sqlparser with lemon-rs, which has up-to-date sqlite3 grammar and working a working code emitter.

close #15

### CODE REVIEW

File: server/src/database/libsql.rs [hunk 1]
```diff
@@ -68,6 +66,70 @@ macro_rules! ok_or_exit {
     };
 }
 
+fn open_db(
```
[the remark this answers is not in the document]
Comment (MarinPostma) [reply server/src/database/libsql.rs:69]: no it's just a move, the block was way to big and I was getting confused

[the remark this answers is not in the document]
Comment (MarinPostma) [reply server/src/database/libsql.rs:69]: nothing changed, just moved away for visibitly

**Comment by psarna** [comment]
I have two doubts:
1. whether `sqlite3-parser` isn't *too* bleeding edge for us, stats from crates.io suggest that we're more or less the only users of the crate :innocent:
2. There's a subtle regression - I'm no longer able to call statements like `PRAGMA` - they now return an "unsupported statement" error. Perhaps we just don't want to allow them, but some of them, like `PRAGMA page_size` are harmless and read-only, and can be used to tune your app based on your database config. Perhaps we can bring the support back?

**Comment by psarna** [comment]
*doubt (1.) is non-blocking, I don't mind switching to sqlite3-parser, we can always go back.

**Comment by penberg** [comment]
@psarna I think we might want to support _some_ harmless pragmas to make switching between local and remote more transparent, but I think the regression is OK for now.

**Comment by penberg** [comment]
bors merge

**Pull request merged**